reconcile/utils/slack_api.py

```diff
@@ -1,12 +1,18 @@
+import os
 import time
 import logging
+from typing import Dict
 
 from slackclient import SlackClient
 from sretoolbox.utils import retry
 
 from reconcile.utils.secret_reader import SecretReader
 from reconcile.utils.config import get_config
 
+# Default result sizes to return from APIs
+CHANNELS_LIMIT = os.environ.get('SLACK_API_CHANNELS_LIMIT', 1000)
```

Comment:
i'm not sure how we'd override these environment variables. in any case, will it be possible to pass them through app-interface? maybe enhance the slack workspace schema?

Reply:
I'm mostly implementing that to keep the option open without changing code in qontract-reconcile.  Environment variables seemed to be a common pattern for handling this in this project.  In reality, it is probably unlikely that we'll need to set them.  Even if Slack API lowered their `limit`, the current behavior is to ignore the value higher than the limit in favor of the limit, without any errors.

Reply:
I discussed this further with @maorfr and this wouldn't be the way to handle the configuration.  I'll remove this for now to get the initial fix out to reduce the run time, but I'll follow-up with another PR for handling the configuration via app-interface.